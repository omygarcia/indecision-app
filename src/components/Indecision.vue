<template>
  <h1>Indecision</h1>
  <img v-if="img" :src="img" alt="bg" />
  <div class="bg-dark"></div>
  <div class="indecision-container">
    <input 
    type="text" 
    placeholder="Hazme una pregunta"
    v-model="question"
     />
    <p>Recuerda terminar con un signo de interrogaci√≥n(?)</p>
    <div v-if="isValidQuestion">
      <h2>{{question}}</h2>
      <h1>{{traducir(answer)}}</h1>
    </div>
  </div>
</template>

<script>

export default {
  name:'Indecision',
  data(){
    return{
      question:null,
      answer:null,
      //img:'https://via.placeholder.com/250'
      img:null,
      isValidQuestion:false
    }
  },
  methods:{
    async getAnswer(){
      this.answer = 'pensando...';
      const data = await fetch('https://yesno.wtf/api').then(r=>r.json());
      console.log(data);
      this.answer = data.answer;
      this.img = data.image;
    },
    traducir(answer)
    {
      switch(answer){
        case 'yes':
          return 'Si!';
        case 'no':
          return 'No!';
        case 'maybe':
          return 'Talvez!';
        default:
          return answer;
      }
    }
  },
  watch:{
    question(value,oldValue){
      //console.log({value,oldValue});
      this.isValidQuestion = false;
      if(!value.includes('?')) return;
      //TODO: hacer peticion http
      this.isValidQuestion = true;
      this.getAnswer();
    }
  }
}
</script>

<style scoped>

    img, .bg-dark {
        height: 100vh;
        left: 0px;
        max-height: 100%;
        max-width: 100%;
        position: fixed;
        top: 0px;
        width: 100vw;
    }

    .bg-dark {
        background-color: rgba(0, 0, 0, 0.4);
    }

    .indecision-container {
        position: relative;
        z-index: 99;
    }
    
    input {
        width: 250px;
        padding: 10px 15px;
        border-radius: 5px;
        border: none;
    }
    input:focus {
        outline: none;
    }

    p {
        color: white;
        font-size: 20px;
        margin-top: 0px;
    }

    h1, h2 {
        color: white;
    }
    
    h2 {
        margin-top: 150px;
    }

</style>