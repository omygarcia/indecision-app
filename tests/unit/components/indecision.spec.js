import { shallowMount } from "@vue/test-utils";
import Indecision from "@/components/Indecision";


describe('Indecision Component',()=>{
    let wrapper = null;

    beforeEach(()=>{
        wrapper = shallowMount(Indecision);
    });

    test('debe hacer match con el snapts', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })
    
});