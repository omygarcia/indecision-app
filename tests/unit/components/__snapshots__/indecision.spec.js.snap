// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indecision Component debe hacer match con el snapts 1`] = `
<h1>Indecision</h1>
<!--v-if-->
<div class="bg-dark"></div>
<div class="indecision-container"><input type="text" placeholder="Hazme una pregunta">
  <p>Recuerda terminar con un signo de interrogaci√≥n(?)</p>
  <!--v-if-->
</div>
`;
